Closing previous PR and creating a new one.

bamartinezd opened the pull request:
Includes Training algorithms and Amazon assessment.

bamartinezd committed:
Cleaning the solution!

bamartinezd committed:
Migrating (.csproj) project files from netcore 3.1 to dotnet 6

Commit:
Some practices about C# programming

Commit:
Adding a new project to the solution

Commit:
Adding test codility

Commit:
Adding directory TiendaPagoCodilityTest

bamartinezd committed:
Firs version of challenge

Commit:
TiendaPagoTest completed

Commit:
Merge and update origin refs

Commit:
Updating some algoritms

bamartinezd committed:
Doing HR divisible sum pairs algo

bamartinezd committed:
New algos

Commit:
Resolving word parser algorithm

Commit:
Adding Algorithn QuickSort

bamartinezd committed:
Migration of previous projects to net7 (.csproj)

Commit:
git commit Migration ArrayManipulation/Printandcountduplicates

bamartinezd committed:
Adding a new project

Commit:
Chanllenges for preparation to Amazon assesment

Commit:
update over the project find-the-running-median

DavidLozada22 committed:
Some updates to the Swap nodes algorithm

DavidLozada22 committed:
Updating swapNodes method

bamartinezd committed:
New challengues agregated

DavidLozada22 committed:
just removing unnecesary assignment

bamartinezd committed:
Resolving comments of PR about MigratoryBirds project

bamartinezd committed:
Context of wordparserIntertec added

andrescollazosc commented on Amazon Transaction Logs/Program.cs:
```diff
@@ -0,0 +1,24 @@
+        List<string> result = processLogs(logs, threshold);
+    }
+
+    public static List<string> processLogs(List<string> logs, int threshold)
```
- This Method is not working, Please remove it.
 - Please read the documentation Code style to c# (https://google.github.io/styleguide/csharp-style.html).

bamartinezd replied:
Resolved, removing the duplicated project from the solution.

andrescollazosc commented on Amazon Transaction Logs/Program.cs:
```diff
@@ -0,0 +1,24 @@
+    {
+        int logsCount = Convert.ToInt32(Console.ReadLine().Trim());
+
+        List<string> logs = new List<string>();
```
Replace List<string> logs = new List<string>(); By var logs = new List<string>();

bamartinezd replied:
Resolved, removing the duplicated project from the solution.

andrescollazosc commented on AmazonDeliveryCenters/Program.cs:
```diff
@@ -0,0 +1,35 @@
+    public static void Main(string[] args)
+    {
+        //var a = suitableLocations(new List<int> { 2, 0, 3, -4 }, 22);
+        //var a = suitableLocations(new List<int> { -2, 1, 0 }, 8);
```
Remove Code that is Not being used.

bamartinezd replied:
So far, we need them as possible scenarios in order to test this method, but we are considering creating unit tests.

andrescollazosc commented on AmazonDeliveryCenters/Program.cs:
```diff
@@ -0,0 +1,35 @@
+    {
+        //var a = suitableLocations(new List<int> { 2, 0, 3, -4 }, 22);
+        //var a = suitableLocations(new List<int> { -2, 1, 0 }, 8);
+        var a = suitableLocations(new List<int> { -1, 0, 1 }, 8);
```
Please, Read Docs C# Style ([List<string> logs = new List<string>();](https://google.github.io/styleguide/csharp-style.html)), changed a By a descriptive word.

bamartinezd replied:
Resolved, replacing the a by suitableCenters.

andrescollazosc commented on AmazonTransactionLogs/Program.cs:
```diff
@@ -0,0 +1,62 @@
+
+        int threshold = 2;
+
+        List<string> result = processLogs(logs, threshold);
```
- please replace List<string> By var.

bamartinezd replied:
.l. 
https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md

10. We only use var when the type is explicitly named on the right-hand side...

andrescollazosc commented on AmazonTransactionLogs/Program.cs:
```diff
@@ -0,0 +1,62 @@
+    {
+        var logs = new List<string> { "1 2 50", "1 7 70", "1 3 20", "2 2 17" };
+
+        int threshold = 2;
```
Please add const int threshold = 2;

andrescollazosc commented on AmazonTransactionLogs/Program.cs:
```diff
@@ -0,0 +1,62 @@
+        List<string> result = processLogs(logs, threshold);
+    }
+
+    public static List<string> processLogs(List<string> logs, int threshold)
```
Please change Name processLogs By ProcessLogs. Read documentation(https://google.github.io/styleguide/csharp-style.html)

andrescollazosc commented on AmazonTransactionLogs/Program.cs:
```diff
@@ -0,0 +1,62 @@
+
+            
+        });
+
```
Please refactor or split this code in another method.

andrescollazosc commented on AmazonTransactionLogs/Program.cs:
```diff
@@ -0,0 +1,62 @@
+            
+        });
+
+        var a = resultIds.GroupBy(id => id)
```
Replace character a By Descriptive Word

andrescollazosc commented on AmazonTransactionLogs/Program.cs:
```diff
@@ -0,0 +1,62 @@
+        List<string> finalIds = new List<string>();
+        foreach (var keyValue in a)
+        {
+            if (keyValue.Count>=threshold)
```
Please Read doc (https://google.github.io/styleguide/csharp-style.html)

andrescollazosc commented on BookSearchPAge/Program.cs:
```diff
@@ -0,0 +1,47 @@
+{
+    public static void Main(string[] args)
+    {
+        var a = SearchPage(6, 4);
```
- Variable a is not used.
 - Replace character a by a descriptive word.

andrescollazosc commented on BookSearchPAge/Program.cs:
```diff
@@ -0,0 +1,47 @@
+        var a = SearchPage(6, 4);
+    }
+
+    public int SearchPage(int n, int p)
```
Please check all this method.

andrescollazosc commented on BookSearchPAge/Program.cs:
```diff
@@ -0,0 +1,47 @@
+    }
+}
+
+class Page
+{
```
Please separate this class in another file

andrescollazosc commented on BruteForceAlgorithm/Program.cs:
```diff
@@ -0,0 +1,66 @@
+        watch.Start();
+
+        DelBruteForceAlgo testCallback = delegate (ref char[] inputs) {
+            var str = new String(inputs);
```
Please changed variable name str by a descriptive name.

andrescollazosc commented on BruteForceAlgorithm/Program.cs:
```diff
@@ -0,0 +1,66 @@
+            var str = new String(inputs);
+            Console.WriteLine($"testCallback - {str}");
+            tries++;
+            return (str == "f71d05c0");
```
Please include this value 'f71d05c0'  in a constant variable

andrescollazosc commented on BruteForceAlgorithm/Program.cs:
```diff
@@ -0,0 +1,66 @@
+            return (str == "f71d05c0");
+        };
+
+        bool result = BruteForce("01234567abcdef", 1, 8, testCallback);
```
Please check result variable, Is not used it.

andrescollazosc commented on ConsoleApp_fit_point74/Program.cs:
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace ConsoleApp_fit_point74
```
Please read documentation

andrescollazosc commented on ConsoleApp_fit_point74/Program.cs:
```diff
@@ -0,0 +1,31 @@
+            return reversed;
+        }
+    }
+}
```
LGTM

andrescollazosc commented on DivisibleSumPairs/Program.cs:
```diff
@@ -0,0 +1,24 @@
+        Console.WriteLine(res);
+    }
+
+    private static int divisibleSumPairs(int n, int k, List<int> ar)
```
Please read documentation

andrescollazosc replied:
define variable names

andrescollazosc commented on DivisibleSumPairs/Program.cs:
```diff
@@ -0,0 +1,24 @@
+            for (int j = i + 1; j < ar.Count; j++)
+            {
+                int res = ar[i] + ar[j];
+                if (res % k == 0)
```
Pleas read documentation.

DavidLozada22 committed:
Refatoring Amazon test algorithms

bamartinezd committed:
Refactoring

bamartinezd committed:
Merged Amazon exercises

bamartinezd committed:
Adding a new challenge from CornerStone, it's a simple algorithm that print a list by a specific numer of elements per row.

bamartinezd committed:
Refactoring

bamartinezd committed:
feat(extra-long-factorials):Add new recursive approach to calculate factorials

 - Currently running extraLongFactorials function.
 - Rename extraLongFactorials function.

Refs: US-AMD

bamartinezd committed:
Refactor(extra-long-factorials):Refactor extraLongFactorials function.

Refs: US-AMD01
